examples/layerpot.py

```diff
@@ -116,9 +121,16 @@ def apply_op(density):
     if enable_mayavi:
         # {{{ plot boundary field
 
-        fld_on_bdry = bound_bdry_op(queue, sigma=sigma, k=k).get()
+        from meshmode.dof_array import flatten
+        from pytools.obj_array import obj_array_vectorize
+
+        fld_on_bdry = actx.to_numpy(
+                flatten(bound_bdry_op(actx, sigma=sigma, k=k)))
+
+        nodes_host = obj_array_vectorize(
+                actx.to_numpy,
+                thaw(actx, density_discr.nodes()))
```

Comment:
Not sure. There aren't a lot of places in the code that call `obj_array_vectorize(actx.to_numpy, ...)`. Two potential issues are:
- calling `actx.to_numpy` on DOFArrays is the wrong thing to do (should flatten first) and should be flagged as an error if vectorizing is allowed
- most of the time, I think a multidimensional NumPy array is wanted instead of an object array of NumPy arrays.

Reply:
I actually added a `flatten_to_numpy` function in `pytential.utils` that flattens more or less everything. 
* It popped up in quite a few places in the `matrix` code because that mostly works with flat numpy arrays.
* It was also useful in porting tests, e.g. when plotting with `matplotlib`.

Not advocating for it in particular, since there may be better ways to do that.

Reply:
Thanks for your thoughts. Let's leave it as it is for now. I agree that it has the potential to be confusing, and that there are a number of possible behaviors.